### ISSUE TITLE
Python 3

### ISSUE THREAD

**Issue by BliNeR-KeY** [su1s/e2m3u2bouquet#127]
hello have have plan to migrate PY3 ?

**Comment by DougMac** [comment]
If there's an image that works on the tmnanosem2plus and uses Python 3 I can have a look and test the changes needed.

I normally use OpenATV but there's not yet a supported image for 6.5 for my box

**Comment by stingray82** [comment]
A Python 3 image is now available for testing from Openvix from here, http://images.satellites-xml.org/Py3/

**Comment by lincsat** [comment]
I believe ATV now have releases of PY3 images. I've tried on ViX and got some of the changes done but am only a Python novice

**Comment by lincsat** [comment]
Does anyone know the replacements for depreciated items AppUrlOpener & FancyURLopener

**Comment by stingray82** [comment]
Lincsat, I think you can use just requests or urllib2 is there a repo your working on this from?

https://pypi.org/project/requests/ & https://docs.python.org/2/library/urllib2.html

**Comment by lincsat** [comment]
All I have done is to run the existing .py though the 2to3 converter and then corrected what syntax errors remain. I am a total novice and haven't looked at changing the code, just converting what I can

**Comment by neoatomic** [comment]
This release works with Py3 :
 https://github.com/oe-mirrors/e2m3u2bouquet-plugin

**Comment by lincsat** [comment]
I've tried that and still get the depreciation warning and it fails to download any M3U's. Maybe it's just the version of PY3 in the latest ViX images?

`DeprecationWarning: AppUrlOpener style of invoking requests is deprecated. Use newer urlopen functions/methods
  urllib._urlopener = AppUrlOpener()`

**Comment by stingray82** [comment]
I tested this in the latest VIX and it appears to work fine, when running it from the command line i get the same error message but the bouquets where generated

**Comment by lincsat** [comment]
Do you use config based setup? I've just tried again with latest ViX developer image and again, doesn't download

```
root@zgemmah7:/usr/lib/enigma2/python/Plugins/Extensions/E2m3u2bouquet# ./e2m3u2bouquet.py
/usr/lib/enigma2/python/Plugins/Extensions/E2m3u2bouquet/./e2m3u2bouquet.py:1617: DeprecationWarning: AppUrlOpener style of invoking requests is deprecated. Use newer urlopen functions/methods
  urllib._urlopener = AppUrlOpener()

********************************
Starting Enigma2 IPTV bouquets v0.8.5
2021-09-25 16:35:10.099711
********************************


********************************
E2m3u2bouquet - Config based setup
********************************


********************************
Config based setup - Test1
********************************


Test1: ----Downloading m3u file----
Test1: Unable to download m3u file from url

********************************
Config based setup - Test2
********************************


Test2: ----Downloading m3u file----
Test2: Unable to download m3u file from url

----Reloading bouquets----
bouquets reloaded...

********************************
Enigma2 IPTV bouquets created !
********************************

```

**Comment by BliNeR-KeY** [comment]
Fixed Plugin on Py2/3
https://github.com/BliNeR-KeY/e2m3u2bouquet-plugin

**Comment by lincsat** [comment]
I've just tried this version and it works when downloading from the XC panel or the M3U on a local http server but not from a local file or direct from a site like m3u4u.

I'm guessing the m3u4u site is giving the flood protection, is the script using multiple connections to trigger that? Is it possible to configure a local file download rather than just a http? The original version did download from m3u4u OK, even multiple servers.

**Comment by KittyBua** [comment]
Here my fix for pli Dev ! Version 0.8.6 

https://github.com/KittyBua/e2m3u2bouquet-0.8.6-pli-dev
